test/e2e/dra/test-driver/app/kubeletplugin.go

```diff
@@ -555,3 +577,56 @@ func (ex *ExamplePlugin) CountCalls(methodSuffix string) int {
 func (ex *ExamplePlugin) UpdateStatus(ctx context.Context, resourceClaim *resourceapi.ResourceClaim) (*resourceapi.ResourceClaim, error) {
 	return ex.kubeClient.ResourceV1beta1().ResourceClaims(resourceClaim.Namespace).UpdateStatus(ctx, resourceClaim, metav1.UpdateOptions{})
 }
+
+// getInfoInterceptor is an interceptor that blocks GetInfo GRPC calls.
+// GetInfo is used by plugin watcher to check if plugin is alive, so blocking it is equivalent to plugin stuckness.
+func (ex *ExamplePlugin) getInfoInterceptor(ctx context.Context, req interface{}, info *grpc.UnaryServerInfo, handler grpc.UnaryHandler) (interface{}, error) {
+	if info.FullMethod == "/pluginregistration.Registration/GetInfo" {
+		ex.blockGetInfoMutex.Lock()
+		defer ex.blockGetInfoMutex.Unlock()
+	}
+	return handler(ctx, req)
+}
+
+func (ex *ExamplePlugin) IsConnected() bool {
+	return ex.connected.Load()
+}
+
+type statsHandler struct {
+	connected *atomic.Bool
+}
+
+func (ex *ExamplePlugin) newStatsHandler() stats.Handler {
+	return &statsHandler{connected: &ex.connected}
+}
+
+// grpc.stats.Handler interface method.
+// HandleConn is called when a GRPC connection starts and ends.
```

Comment:
Is this mechanism reliable, in particular the "connection ends" part?

If yes, do we need the application level pings at all?

If no, then `h.connected.Store(false)` cannot be relied upon for testing.

Reply:
It's reliable on the GRPC server side. At least I didn't see it being unreliable. I'm not sure we can use it on the Kubelet side, at least without changing pluginmanager connections to be persistent. Current implementation closes them.